src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -2790,7 +2790,11 @@ private void CompileTypeEqualExpression(Expression expr)
             var node = (TypeBinaryExpression)expr;
 
             Compile(node.Expression);
-            if (node.TypeOperand.GetTypeInfo().IsGenericType && node.TypeOperand.GetGenericTypeDefinition() == typeof(Nullable<>))
+            if (node.Expression.Type == typeof(void))
+            {
+                _instructions.EmitLoad(node.TypeOperand == node.Expression.Type, typeof(bool));
+            }
+            else if (node.TypeOperand.GetTypeInfo().IsGenericType && node.TypeOperand.GetGenericTypeDefinition() == typeof(Nullable<>))
```

Comment:
Wondering whether we have the `IsNullableType` extension method from `TypeUtils` available to simplify these checks?